### PR TITLE
Atexit Functionality To Close DB Connections 

### LINKED ISSUE #458: close all connections when shutting down the kernel
currently, users have to manually close connectios via `--close`. But in most cases, it'd be better to close them automatically. for example, when they restart the kernel or they shut down jupyterlab

we can use the atexit module to do that: https://docs.python.org/3/library/atexit.html and close all connections right before the kernel shuts down.

### PR DESCRIPTION
**Pull request by TegveerG**
## Describe your changes

Created new branch with updated pull (no reverts to existing files). 
Code same as before (#547) 

## Issue number

Closes #458 

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--551.org.readthedocs.build/en/551/

### CODE REVIEW

**Comment by edublancas** [comment]
is this ready for review? you're missing unit tests for the close connections function

**Comment by TegveerG** [comment]
> is this ready for review? you're missing unit tests for the close connections function

Hi, adding unit test and pushing a final commit. Not ready for review yet.

**Comment by TegveerG** [comment]
@edublancas ready for review!

**Review by edublancas — CHANGES_REQUESTED** [review]
coincidentally, another PR already implemented a `close_all` function that's been unit tested so for this PR you only need to add the `atexit.register` line

File: src/tests/test_connection.py [hunk 1]
```diff
@@ -305,3 +305,32 @@ def close(self):
 def test_custom_connection(conn, expected):
     is_custom = Connection.is_custom_connection(conn)
     assert is_custom == expected
+
+
+def test_close_all_connections(cleanup):
```
[the remark this answers is not in the document]
Comment (edublancas) [reply src/tests/test_connection.py:310]: also, please fix the merging conflicts

**Pull request closed without merging**